Libplanet/Blocks/BlockContent.cs

```diff
@@ -195,22 +195,29 @@ public IReadOnlyList<Transaction<T>> Transactions
         }
 
         /// <summary>
-        /// Mines the PoW (proof-of-work) nonce satisfying the block
-        /// <see cref="BlockMetadata.Difficulty"/>, and returns a valid
-        /// <see cref="PreEvaluationBlock{T}"/> instance.
+        /// Mines the PoW (proof-of-work) nonce satisfying <paramref name="difficulty"/>
+        /// for <see cref="PreEvaluationBlockHeader.PreEvaluationHash"/>
+        /// and returns a valid <see cref="PreEvaluationBlock{T}"/> instance.
         /// </summary>
+        /// <param name="difficulty">The difficulty to target when mining
+        /// <see cref="PreEvaluationBlockHeader.PreEvaluationHash"/>.</param>
         /// <param name="cancellationToken">An optional cancellation token used to propagate signal
         /// that this operation should be cancelled.</param>
         /// <returns>A <see cref="PreEvaluationBlock{T}"/> instance with a valid proof-of-work.
         /// </returns>
         /// <exception cref="OperationCanceledException">Thrown when the specified
         /// <paramref name="cancellationToken"/> received a cancellation request.</exception>
-        public PreEvaluationBlock<T> Mine(CancellationToken cancellationToken = default) =>
-            new PreEvaluationBlock<T>(this, MineNonce(cancellationToken).PreEvaluationHash);
+        public PreEvaluationBlock<T> Mine(
```

Comment:
This method will be removed eventually?

Reply:
Probably? I'm trying to keep it as long as possible to to enable mock testing. Perhaps I should move it tests at some point. 🤔